cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -105,50 +141,40 @@ func enforceRequirements(flags *applyPlanFlags, args []string, dryRun bool, upgr
 			// This also makes the KubernetesVersion value returned for `upgrade plan` consistent as that command
 			// allows to not specify a target version in which case KubernetesVersion will always hold the currently
 			// installed one.
-			cfg.KubernetesVersion = newK8sVersion
+			initCfg.KubernetesVersion = newK8sVersion
 		}
 	}
 
-	ignorePreflightErrorsSet, err := validation.ValidateIgnorePreflightErrors(flags.ignorePreflightErrors, cfg.NodeRegistration.IgnorePreflightErrors)
-	if err != nil {
-		return nil, nil, nil, err
-	}
-	// Also set the union of pre-flight errors to InitConfiguration, to provide a consistent view of the runtime configuration:
-	cfg.NodeRegistration.IgnorePreflightErrors = sets.List(ignorePreflightErrorsSet)
-
-	// Ensure the user is root
-	klog.V(1).Info("running preflight checks")
-	if err := runPreflightChecks(client, ignorePreflightErrorsSet, printer); err != nil {
-		return nil, nil, nil, err
-	}
```

Comment:
this diff is necessary, I must move this code block up before the call to `FetchInitConfigurationFromCluster`, so that the interaction with cluster will not happen, this is because some testcases want to check the logic around this `runPreflightChecks`.

see this testcase:
cmd/kubeadm/app/cmd/upgrade TestEnforceRequirements/Fail_pre-flight_check

and the log from here: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/114062/pull-kubernetes-unit/1715315999886020608/build-log.txt